nexus/types/src/deployment/execution/dns.rs

```diff
@@ -132,6 +132,27 @@ pub fn blueprint_internal_dns_config(
         )?;
     }
 
+    // For each sled to which we are supposed to be replicating artifacts,
+    // define DNS entries for the repo depot service.
+    //
+    // Consumers need to be careful in using these names since artifacts are not
+    // replicated synchronously or atomically to all instances.  That is: a
+    // consumer should be careful when fetching an artifact about whether they
+    // really can just pick any backend of this service or not.
+    for (sled_id, sled) in sleds_by_id {
+        if sled.policy().matches(SledFilter::TufArtifactReplication) {
+            continue;
+        }
```

Comment:
Isn't this the opposite of what we want?

Reply:
Ugh, yes -- sorry, I'm still in the process of testing this.